smart-contracts/contracts/cl-vault/src/helpers.rs

```diff
@@ -231,13 +263,17 @@ pub fn assert_admin(deps: Deps, caller: &Addr) -> Result<Addr, ContractError> {
 }
 
 pub fn round_up_to_nearest_multiple(amount: i64, multiple: i64) -> i64 {
+    if amount.abs() < multiple {
+        return 0; // Return 0 when amount is less than multiple
+    }
+
     let remainder = amount % multiple;
     if remainder == 0 {
-        amount
+        amount // Amount is already a multiple of 'multiple'
     } else if amount < 0 {
-        amount - remainder
+        amount + remainder.abs() // Round down to the nearest multiple
```

Comment:
This also looks effectively round up if amount < 0, as we are adding remainder.abs()  to a negative number.